src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1442,6 +1446,129 @@ def disable_features(cmd, client, resource_group_name, cluster_name, features, k
+
+        try:
+            # setting kubeconfig
+            kube_config = set_kube_config(kube_config)
+            # Loading the kubeconfig file in kubernetes client configuration
+            load_kube_config(kube_config, kube_context)
+            configuration = kube_client.Configuration()
+        except Exception as ex:
+            logger.error("Couldn't load kubeconfig. Error: {}".format(str(ex)))
+            tr_logger.error("Couldn't load kubeconfig. Error: {}".format(str(ex)))
+
+        # Install helm client
+        helm_client_location = install_helm_client()
+
+        try:
+            validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context, helm_client_location)
+        except Exception as e:
+            logger.error(str(e))
+            tr_logger.error(str(e))
+
+        kapi_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+        try:
+            utils.try_list_node_fix()
+            api_response = kapi_instance.list_node()
+            required_node_counts = 0
+            for item in api_response.items:
+                node_arch = item.metadata.labels.get("kubernetes.io/arch")
+                node_os = item.metadata.labels.get("kubernetes.io/os")
+                if node_arch == "amd64" and node_os == "linux":
+                    tr_logger.info("Node {} has capacity: CPU: {}, and Memory: {} ".format(required_node_counts + 1, item.status.capacity.get("cpu"), item.status.capacity.get("memory")))
+                    required_node_counts += 1
+            if required_node_counts == 0:
+                tr_logger.error("There are no linux/amd64 nodes available on the kubernetes cluster.")
+                logger.error("This kubernetes cluster doesn't have any nodes with OS 'linux' and architecture 'amd64', for scheduling the Arc-Agents onto and connecting to Azure. Learn more at {}".format("https://aka.ms/ArcK8sSupportedOSArchitecture"))
+        except Exception as ex:
+            tr_logger.error("Couldn't list nodes on the k8s cluster. Error: {}".format(str(ex)))
+
+        try:
+            pod_list = kapi_instance.list_namespaced_pod(consts.Arc_Namespace)
+            pods_count = 0
+            pods_pending = 0
+            for pod in pod_list.items:
+                pods_count += 1
+                tr_logger.debug("Pod {} is in {} state. Reason: {}. Container statuses: {}".format(pod.metadata.name, pod.status.phase, pod.status.reason, pod.status.container_statuses))
+                if pod.status.phase == "Pending":
+                    pods_pending += 1
+            if pods_pending > 0:
+                logger.warning("You have pods stuck in pending state. A possible cause is insuffient resources on the Kubernetes cluster you are trying to onboard to azure-arc. Please run kubectl get pods -n azure-arc to ensure all pods are in running state.")
```

Comment:
CLI output should read:

**"Some pods in the azure-arc namespace are stuck in a pending state. Insufficient resources may sometimes cause this. Please run kubectl get pods -n azure-arc to ensure your pods are in a running state."**

Color: Red

Reply:
done